### PR TITLE
Type component tokens final

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
This PR types all the component design tokens.

Includes the four already reviews and approved PRs:
https://github.com/aws-amplify/amplify-ui/pull/1587
https://github.com/aws-amplify/amplify-ui/pull/1593
https://github.com/aws-amplify/amplify-ui/pull/1594
https://github.com/aws-amplify/amplify-ui/pull/1595

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/theme/tokens/components/checkbox.ts [hunk 1]
```diff
@@ -1,4 +1,104 @@
...
+interface DisableToken {
+  cursor: DesignToken<CursorValue>;
+}
+
+interface ButtonDisabledToken {
+  borderColor: DesignToken<BorderColorValue>;
+}
+interface ButtonErrorFocusToken {
+  borderColor: DesignToken<BorderColorValue>;
+  boxShadow: DesignToken<ShadowValue>;
```
Comment (jacoblogan) [inline packages/ui/src/theme/tokens/components/checkbox.ts:36]:
```suggestion
  boxShadow: DesignToken<BoxShadowValue>;
```

File: packages/ui/src/theme/tokens/components/fieldControl.ts [hunk 2]
```diff
@@ -1,4 +1,105 @@
-export const fieldcontrol = {
+import {
+  BackgroundColorValue,
```
Comment (jacoblogan) [inline packages/ui/src/theme/tokens/components/fieldControl.ts:2]:
```suggestion
  BackgroundColorValue,
  BorderColorValue,
```

File: packages/ui/src/theme/tokens/components/radio.ts [hunk 3]
```diff
@@ -1,4 +1,106 @@
-export const radio = {
+import {
+  AlignItemsValue,
+  BackgroundColorValue,
```
Comment (jacoblogan) [inline packages/ui/src/theme/tokens/components/radio.ts:3]:
```suggestion
  BackgroundColorValue,
  BorderColorValue,
```

Comment (reesscot) [reply packages/ui/src/theme/tokens/components/radio.ts:3]: good catch, fixed.

File: packages/ui/src/theme/tokens/components/fieldControl.ts (hunk 2 above)
Comment (reesscot) [reply packages/ui/src/theme/tokens/components/fieldControl.ts:2]: fixed

**Review by 0618 — APPROVED** [review]

**Review by jacoblogan — APPROVED** [review]
Just one last nit, but otherwise LGTM!

File: packages/ui/src/theme/tokens/components/switchField.ts [hunk 4]
```diff
@@ -1,4 +1,77 @@
...
+  TransformValue,
+  TransitionDurationValue,
+} from '../types/designToken';
+
+interface SwitchFieldDisabledTokens {
+  opacity: DesignToken<OpacityValue>;
+}
+
+interface SwitchFieldFocusedTokens {
+  shadow: DesignToken<BoxShadowValue>;
```
[the remark this answers is not in the document]
Comment (reesscot) [reply packages/ui/src/theme/tokens/components/switchField.ts]: fixed

**Pull request merged**